src/raftstore/store/store.rs

```diff
@@ -1039,16 +1173,20 @@ impl<T: Transport, C: PdClient> Store<T, C> {
             self.raft_metrics.message_dropped.region_no_peer += 1;
             return Ok(Some(key));
         }
-        if let Some((_, &exist_region_id)) = self.region_ranges
+
+        let r = self.region_ranges
```

Comment:
How about write it as a function? So that line 776 can also use that.

Reply:
I don't think they are the same and may be much longer than expected after rewritten.